src/snowcli/cli/nativeapp/run_processor.py

```diff
@@ -0,0 +1,214 @@
+
+            # 2. Check for an existing application by the same name
+            show_app_row = self.get_existing_app_info()
+
+            # 3. If existing application is found, perform a few validations and upgrade the instance.
+            if show_app_row is not None:
+
+                # Check if not created by snowCLI or not created using "loose files" / stage dev mode.
+                if show_app_row[COMMENT_COL] != SPECIAL_COMMENT or (
+                    show_app_row[VERSION_COL] != LOOSE_FILES_MAGIC_VERSION
+                ):
+                    raise ApplicationAlreadyExistsError(self.app_name)
+
+                # Check for the right owner
+                ensure_correct_owner(
+                    row=show_app_row, role=self.app_role, obj_name=self.app_name
+                )
+
+                # If all the above checks are in order, proceed to upgrade
+                try:
+                    if diff.has_changes():
+                        print(f"Upgrading existing application {self.app_name}.")
+                        self._execute_query(
+                            f"alter application {self.app_name} upgrade using @{self.stage_fqn}"
+                        )
+
+                    # ensure debug_mode is up-to-date
+                    self._execute_query(
+                        f"alter application {self.app_name} set debug_mode = {self.debug_mode}"
+                    )
+
+                    return
+
+                except ProgrammingError as err:
+                    _generic_sql_error_handler(err)
+
+            # 4. If no existing application is found, create an app using "loose files" / stage dev mode.
+            print(f"Creating new application {self.app_name} in account.")
+
+            if self.app_role != self.package_role:
+                with self.use_role(new_role=self.package_role):
+                    self._execute_queries(
+                        dedent(
+                            f"""\
+                        grant install, develop on application package {self.package_name} to role {self.app_role};
+                        grant usage on schema {self.package_name}.{self.stage_schema} to role {self.app_role};
+                        grant read on stage {self.stage_fqn} to role {self.app_role};
+                        """
+                        )
+                    )
+
+            stage_name = StageManager.quote_stage_name(self.stage_fqn)
+
+            try:
+                self._execute_query(
+                    dedent(
+                        f"""\
+                    create application {self.app_name}
+                        from application package {self.package_name}
+                        using {stage_name}
+                        debug_mode = {self.debug_mode}
+                        comment = {SPECIAL_COMMENT}
+                    """
+                    )
+                )
+            except ProgrammingError as err:
+                _generic_sql_error_handler(err)
```

Comment:
This method is pretty long and consists of three distinct steps:
- validate warehouse existence
- application update
- application creation 

Extracting each step into separate private methods could improve readability.
Something like this:
- create dedicated methods for new app and updating existing.
- invoke methods based on result of `show_app_row = self.get_existing_app_info()`
- 
```python
def _create_new_app(): 
    ...
def _update_existing_app(): 
    ...

def _create_dev_app(self, ...):
    # warehouse checks
    ....
    show_app_row = self.get_existing_app_info()
    if show_app_row:
         self._update_existing_app(show_app_row, ...)
    else:
         self._create_new_app(...)
```

Reply:
I agree that this method can be split out into separate private methods and is on my docket. However for scope of this PR, this method has just been moved from a different file to this file. As this PR blocks two other PRs from going in, I'd like to merge this in, if it's okay with you, and address this method in a follow up PR.